src/CAN/can_mapping.c

```diff
@@ -44,10 +49,52 @@ float canmapping_extract_value(uint64_t raw_data, const CANMapping *mapping)
                         raw_value = swap_uint32(raw_value);
                         break;
                     default:
+                        /* Should never be mapping anything larger than 4 bytes */
+                        panic(PANIC_CAUSE_UNREACHABLE);
                         break;
                 }
         }
-        return (float)raw_value;
+
+        /* convert type */
+        switch (mapping->type) {
+            case CANMappingType_unsigned:
+                    return (float)raw_value;
+        	case CANMappingType_signed:
+                    if (length <= 8) {
+                        return (float)*((int8_t*)&raw_value);
+                    }
+                    else if (length <= 16){
```

Comment:
NIT: the `else if` statements here are totally redundant because you are doing direct returns from within the if statements.